### PR TITLE
Add scripts for file upload

### LINKED ISSUE #11: As an User, I am able to upload files to OSP
## Description
As an User, I am able to upload files to OSP

## Mocks
n/a

### Update [Required]
- [ ] Create upload file button
- [ ] Connect endpoints to send file as formdata
- [ ] Allow ability to download files from bucket (permission based)

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
2-3 days

### PR DESCRIPTION
**Pull request by bismitaguha**
### Description

Fixes #11 

### Type of Change:

- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

N/A

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by bismitaguha** [comment]
@abha224 @sidvenu I am having doubts regarding the URL of upload. So I needed help there

**Comment by sidvenu** [comment]
@bismitaguha could you comment the doubts here?

**Review by sidvenu — DISMISSED** [review]
Looks good till the work done.

**Comment by abha224** [comment]
@bismitaguha Is the issue resolved?

**Comment by bismitaguha** [comment]
@abha224 @sidvenu the URL is throwing errors.

**Comment by abha224** [comment]
@bismitaguha can you add a screenshot of the error for reference?

**Comment by bismitaguha** [comment]
URL error
Screenshot from 2020-07-21 23-27-15

**Comment by sidvenu** [comment]
This sounds like you're appending a JS object to a string. Try running the following in your browser console:

```
const obj = {
    name: "John Smith",
    email: "johnsmith@nomail.com"
};
const urlPath = '/' + obj
urlPath
```
This will print `/[object Object]`.

Can you recheck your code for a similar pattern?

**Comment by bismitaguha** [comment]
Screenshot from 2020-08-03 20-27-25
I am constantly receiving this error and have tried modifying the `.env` file also based on these sources [docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html) & [github issue](https://github.com/aws/aws-cli/issues/1037).
If @sidvenu and @abha224 you could look into this and help me out here. I have referred to [this blog](https://medium.com/@only4/upload-files-to-s3-buckets-from-react-45aaaa651e28) while writing the code.

**Comment by sidvenu** [comment]
Update: Bismita and I discussed this in our 1:1, and we found that this is a problem caused by the `.env` file not getting loaded. @bismitaguha is working to solve that.

**Comment by bismitaguha** [comment]
Update: The env problem is solved. Now I am receiving a different error
SignatureDoesNotMatch: The request signature we calculated does not match the signature you provided. Check your key and signing method.

I am working on this, meanwhile if anyone finds any resource please send a link. @sidvenu @abha224

**Comment by bismitaguha** [comment]
The complete config:
Screenshot from 2020-08-04 01-19-45

**Comment by bismitaguha** [comment]
@sidvenu @abha224 All issues are resolved please review!!!!

File: src/components/Upload.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react'
+// import { postUpload } from '../actions/upload'
```
Comment (sidvenu) [inline src/components/Upload.js]: If not necessary, remove.

File: src/components/Upload.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import AWS from 'aws-sdk'
+require('dotenv').config();
+
+export default class Upload extends Component {
+    constructor(props) {
+        super(props)
```
Comment (sidvenu) [inline src/components/Upload.js]: A code comment here would be appropriate.

File: src/urls.js [hunk 3]
```diff
@@ -27,6 +27,10 @@ export function submission() {
     return `${urlBaseFrontend()}submission`
 }
 
+export function file() {
```
Comment (sidvenu) [inline src/urls.js]:
```suggestion
export function urlFileUpload() {
```

Comment (sidvenu) [reply src/urls.js]: Rename the references to this function too.

**Review by sidvenu — CHANGES_REQUESTED** [review]

**Comment by bismitaguha** [comment]
@sidvenu this is the sample script which will be reused in a separate component in the form responses. So this part will get deleted there completely. So right now do I need to add comments here because I will be resuing this in a different component and adding comments there makes sense maybe???

**Comment by sidvenu** [comment]
@bismitaguha In that case, you can copy and paste the comment as well. I feel an individual contributor should be able to understand this repo's code at all points in history, and thus comments are important no matter how redundant it would be in the future. @abha224 what do you think?

**Review by abha224 — APPROVED** [review]
Reviewed and tested. PR looks good to me :+1:

**Review by sidvenu — APPROVED** [review]
LGTM!

**Pull request merged**